pkg/shared/families/infofinder/sshtopology/scanner.go

```diff
@@ -0,0 +1,366 @@
+		return nil, fmt.Errorf("failed to get ssh authorized keys fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh authorized keys fingerprints %+v", infos)
+
+	return infos, nil
+}
+
+func (s *Scanner) getSSHKnownHostsFingerprints(homeUserDir string) ([]types.Info, error) {
+	infos, err := s.getFingerprints([]string{path.Join(homeUserDir, ".ssh/known_hosts")}, types.SSHKnownHosts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ssh known hosts fingerprints: %v", err)
+	}
+	s.logger.Debugf("Found ssh known hosts fingerprints %+v", infos)
+
+	return infos, nil
+}
+
+func (s *Scanner) getFingerprints(paths []string, infoType types.InfoType) ([]types.Info, error) {
+	var infos []types.Info
+
+	for _, p := range paths {
+		_, err := os.Stat(p)
+		if os.IsNotExist(err) {
+			s.logger.Debugf("File (%v) does not exist.", p)
+			continue
+		} else if err != nil {
+			return nil, fmt.Errorf("failed to check file: %v", err)
+		}
+
+		var output []byte
+		if output, err = s.executeSSHKeyGenCommand("sha256", p); err != nil {
+			return nil, fmt.Errorf("failed to execute ssh-keygen command: %v", err)
+		}
+
+		infos = append(infos, parseSSHKeyGenCommandOutput(string(output), infoType, p)...)
+	}
+
+	return infos, nil
+}
+
+func (s *Scanner) getPrivateKeysPaths(rootPath string, recursive bool) ([]string, error) {
+	var paths []string
+	err := filepath.Walk(rootPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			if path != rootPath && !recursive {
+				return filepath.SkipDir
+			}
+			return nil
+		}
+
+		isPrivateKeyFile, err := isPrivateKey(path)
+		if err != nil {
+			s.logger.Errorf("failed to verify if file (%v) is private key file - skipping: %v", path, err)
+			return nil
+		}
+
+		if isPrivateKeyFile {
+			paths = append(paths, path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to walks the file tree rooted at %v: %v", rootPath, err)
+	}
+
+	return paths, nil
+}
+
+func isPrivateKey(path string) (bool, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return false, fmt.Errorf("failed to open file: %v", err)
+	}
+	defer f.Close()
+
+	// Splits on newlines by default.
+	scanner := bufio.NewScanner(f)
+
+	for scanner.Scan() {
+		// We only need to look at the first line
+		return strings.Contains(scanner.Text(), "PRIVATE KEY"), nil
+	}
+
+	if err = scanner.Err(); err != nil {
+		return false, fmt.Errorf("failed to scan file: %v", err)
+	}
+
+	return false, nil
+}
+
+func parseSSHKeyGenCommandOutput(output string, infoType types.InfoType, path string) []types.Info {
+	lines := strings.Split(output, "\n")
+	infos := make([]types.Info, 0, len(lines))
+	for i := range lines {
+		if lines[i] == "" {
+			continue
+		}
+		infos = append(infos, types.Info{
+			Type: infoType,
+			Path: path,
+			Data: lines[i],
+		})
+	}
+	return infos
+}
+
+func (s *Scanner) executeSSHKeyGenCommand(hashAlgo string, filePath string) ([]byte, error) {
```

Comment:
what about using go ssh native package? https://godoc.org/golang.org/x/crypto/ssh#FingerprintSHA256?

Reply:
I've consulted with @Tehsmash and we decided it's good enough to use the `ssh-keygen` command, I'm not sure if the ssh go pkg support creating a fingerprint from a private key file.
If you think it's better I can try to see if it works the same.

Reply:
If it works the same and it removes an external system level dependency then I think it could be worth switching to the native implementation instead of execing but its not much different from how the other families are operating.